server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -290,9 +293,13 @@ long ramBytesUsedQueryResult(QuerySearchResult result) {
             if (hasAggs == false) {
                 return 0;
             }
-            return result.aggregations()
-                .asSerialized(InternalAggregations::readFrom, namedWriteableRegistry)
-                .ramBytesUsed();
+            if (result.aggregations().isSerialized()) {
+                return result.aggregations()
+                    .asSerialized(InternalAggregations::readFrom, namedWriteableRegistry)
+                    .ramBytesUsed();
+            } else {
+                return result.aggregations().expand().getSerializedSize();
+            }
```

Comment:
I find this change hard to follow, I am sure I am missing something. Why do we need the conditional compared to before? And why do we call asSerialized when the aggregation is already serialized?

Reply:
This is just to avoid serializing on a real buffer since we don't need to keep the result if the aggregations is already expanded. However I agree that it can be confusing so I pushed https://github.com/elastic/elasticsearch/pull/66280/commits/c8673076f2de921f4799de19de3e82d2c83e0149 to address your concerns. Does it look better ?

Reply:
yea much better thanks